src/extender.test.js

```diff
@@ -0,0 +1,184 @@
+import * as extender from './extender';
+import shimmer from 'shimmer';
+
+export const DummyCounterService = (() => {
+  let dummyCounter = 0;
+  const incrementToDummyCounter = (count = 1) => {
+    dummyCounter += count;
+  };
+  const incrementToDummyCounterMultipleParam = (count1 = 1, count2 = 1) => {
+    dummyCounter += count1;
+    dummyCounter += count2;
+  };
+  const getDummyCounter = () => dummyCounter;
+  const reset = () => {
+    dummyCounter = 0;
+  };
+  return { incrementToDummyCounter, getDummyCounter, incrementToDummyCounterMultipleParam, reset };
+})();
+
+describe('extender', () => {
+  beforeEach(() => {
+    //clean all shimmers
+    shimmer.unwrap(DummyCounterService, 'incrementToDummyCounter');
+    shimmer.unwrap(DummyCounterService, 'incrementToDummyCounterMultipleParam');
+    shimmer.unwrap(DummyCounterService, 'getDummyCounter');
+    DummyCounterService.reset();
+  });
+
+  test('hook -> simple flow', () => {
+    extender.hook(DummyCounterService, 'incrementToDummyCounter');
+    DummyCounterService.incrementToDummyCounter();
+    expect(DummyCounterService.getDummyCounter()).toEqual(1);
+  });
+
+  test('hook -> before Hook', () => {
+    let beforeCounter = 0;
+    extender.hook(DummyCounterService, 'incrementToDummyCounter', {
+      beforeHook: () => {
+        expect(DummyCounterService.getDummyCounter()).toEqual(0);
```

Comment:
💪  Pro

Reply:
yeah, but here you're not checking aby before/after order...
you can simply replace the `beforeCounter` with `done` and it will be more readable
(Not mandatory)